### PR TITLE
Bugfix/cannot close captcha by click outside

### PR DESCRIPTION
**Pull request by CAMOBAP**
- #28

### CODE REVIEW

File: Example.App.js [hunk 1]
```diff
@@ -13,20 +13,15 @@ export default class App extends React.Component {
   onMessage = event => {
     if (event && event.nativeEvent.data) {
       if (['cancel'].includes(event.nativeEvent.data)) {
-        // do not call this.captchaForm.hide(); function in this condition. Otherwise app will crash.
+        this.captchaForm.hide();
```
Comment (e271828-) [inline Example.App.js:16]: Why is this safe now?

File: MAINTAINER.md [hunk 2]
```diff
@@ -25,6 +25,10 @@ For iOS instead the last step do:
 - `pushd ios; pod install; popd`
 - `yarn ios`
 
+To quick update `react-native-hcaptcha` localy just run:
```
Comment (e271828-) [inline MAINTAINER.md:28]:
```suggestion
To quickly update `react-native-hcaptcha` locally just run:
```

File: index.js [hunk 3]
```diff
@@ -13,11 +13,10 @@ class ConfirmHcaptcha extends PureComponent {
   show = () => {
     this.setState({ show: true });
   };
-  hide = () => {
+  hide = (source) => {
     const { onMessage } = this.props;
     this.setState({ show: false });
-    // In android on hardware back , emits 'cancel' which can be used to hide hCaptcha modal using ref
-    if (Platform.OS === "android") {
+    if (source) { // if source === undefined => called by the user
```
Comment (e271828-) [inline index.js:19]: Why would we not emit cancel when called by user?

File: Example.App.js (hunk 1 above)
Comment (CAMOBAP) [reply Example.App.js:16]: Due to the previous implementation (which was wrong in my opinion), this call may lead to implicit recursion (infinite calls): `hide -> onMessage -> hide -> onMessage ...`

This PR changes it:
 - all `this.captchaForm.hide();` (hide without arguments) considered as calls from "user code" and will not lead to `onMessage` callback
 - calls `hide` with `source` argument like this `this.captchaForm.hide("back_button");` will lead to `onMessage({nativeEvent:{data: 'cancel'}})`

**Comment by beqramo** [comment]
Please merge it I need it
Thanks

**Review by e271828- — APPROVED** [review]

**Pull request merged**